app/models/webhook.rb

```diff
@@ -1,11 +1,17 @@
 class Webhook < ActiveRecord::Base
   has_soft_deletion default_scope: true
-  validates :branch, uniqueness: { scope: [ :stage], conditions: -> { where("deleted_at IS NULL") }, message: "one webhook per (stage, branch) combination." }
+  validates :branch, uniqueness: { scope: [ :stage ], conditions: -> { where("deleted_at IS NULL") }, message: "one webhook per (stage, branch) combination." }
 
   belongs_to :project
   belongs_to :stage
 
+  SOURCES = %w[semaphore tddium travis jenkins buildkite github].freeze
```

Comment:
hmm this is repeating all controller names, would be neat to have a test that makes sure that list is complete